hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -282,6 +282,8 @@ protected void commit(HoodieTable table, String commitActionType, String instant
     writeTableMetadata(table, instantTime, commitActionType, metadata);
     activeTimeline.saveAsComplete(new HoodieInstant(true, commitActionType, instantTime),
         Option.of(metadata.toJsonString().getBytes(StandardCharsets.UTF_8)));
+    // reload active timeline
+    table.getMetaClient().reloadActiveTimeline();
   }
```

Comment:
A better choice is moveing the logic into `HoodieTimelineArchiver` constructor.

Reply:
@danny0405 `table` is as a reference in `HoodieTimelineArchiver` constructor, if move to constructor, then the behavior of `reload` will depend on whether the archiver is construct

How about manually `reload` after the writing is complete? In this way, other subsequent services using `table` can obtain the complete time line

Reply:
@Zouxxyy : Reloading timeline after commit would not be beneficial for cases when archiving is run async as it would induce unnecessary filesystem calls. I also agree with @danny0405 comment asking if this is really needed as the archiver would archive the commit in the next run.